### PR TITLE
Issue#55&56 data packaging & sending

### LINKED ISSUE #55: As a programmer, I need a way of packaging data to send it to my peers so that data isn't lost by poor encoding
acceptance criteria: 
- The InMemoryFile class shall be modified to be packaged and unpackaged for sending via WifiDirect
- A demonstration must occur of the data object being packaged an unpackaged correctly.

### LINKED ISSUE #56: As a programmer, I need to be able to send and receive data between peers so that communication is possible
acceptance criteria:
- the implementation for sending data shall occur in the WifiDirect Singleton
- The data sent and the data received shall be saved to the local Room database
- The sending and receiving functionality must be demonstrable, since instrumented testing is impossible due to the need for multiple devices
- the WifiDirect singleton shall use the modified InMemoryFile class from issue #55 (As a programmer, I need a way of packaging data to send it to my peers so that data isn't lost by poor encoding) to package and unpackage data.

### PR DESCRIPTION
**Pull request by Madministrator**
closes #55 
closes #56 

This meets all of the acceptance criteria accept for the demonstration criteria. If I did that, it would interfere with code in other people's tasks for this sprint. As such, I want to open that in a new branch.

### CODE REVIEW

**Comment by Austin-Scott** [comment]
So you have successfully figured out how to send and receive data?

File: app/src/main/java/edu/uwstout/p2pchat/FileTransfer/UpdaterAsyncTask.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    @Override
+    protected Object doInBackground(final Object[] objects)
+    {
+        try
+        {
+            ServerSocket serverSocket = new ServerSocket(MAGIC_PORT);
+            Log.i(LOG_TAG, "Updater: Socket opened");
+            // The call to accept is blocking, which is why this
+            // is in a separate thread.
+            Socket client = serverSocket.accept();
```
[the remark this answers is not in the document]
Comment (Madministrator) [reply app/src/main/java/edu/uwstout/p2pchat/FileTransfer/UpdaterAsyncTask.java]: They are on separate ports.

File: app/src/main/java/edu/uwstout/p2pchat/FileTransfer/ReceiverAsyncTask.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+ * A simple server socket that accepts connections and writes
+ * some data on the stream.
+ * Only applicable when this device is acting as the server.
+ */
+public class ReceiverAsyncTask extends AsyncTask
+{
+    /**
+     * Application context needed to do tasks.
+     */ // suppress warnings related to context leak.
+    @SuppressLint("StaticFieldLeak")
```
[the remark this answers is not in the document]
Comment (Madministrator) [reply app/src/main/java/edu/uwstout/p2pchat/FileTransfer/ReceiverAsyncTask.java]: it was as simple as hitting alt+enter when the cursor was on the highlighted parts.

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -30,6 +30,8 @@
         <service
             android:name="edu.uwstout.p2pchat.WifiDirect"
             android:exported="false"/>
+        <service android:enabled="true"
+            android:name="edu.uwstout.p2pchat.FileTransfer.SendDataService" />
```
[the remark this answers is not in the document]
Comment (Madministrator) [reply app/src/main/AndroidManifest.xml]: yeah, but only when needed.

**Pull request merged**